backend/danswer/connectors/sharepoint/connector.py

```diff
@@ -0,0 +1,230 @@
+
+UNSUPPORTED_FILE_TYPE_CONTENT = ""  # idea copied from the google drive side of things
+
+
+logger = setup_logger()
+
+def get_text_from_xlsx_driveitem(driveitem_object: DriveItem):
+
+    file_content = driveitem_object.get_content().execute_query().value
+    excel_file = io.BytesIO(file_content)
+    workbook = openpyxl.load_workbook(excel_file, read_only=True)
+
+    full_text = []
+    for sheet in workbook.worksheets:
+        sheet_string = '\n'.join(
+                            ','.join(map(str, row)) 
+                            for row in sheet.iter_rows(min_row=1, values_only=True)
+                        )
+        full_text.append(sheet_string)
+        
+    return '\n'.join(full_text)
+
+def get_text_from_docx_driveitem(driveitem_object: DriveItem):
+    file_content = driveitem_object.get_content().execute_query().value
+    full_text = []
+
+    with tempfile.TemporaryDirectory() as local_path:
+        with open(os.path.join(local_path, driveitem_object.name), "wb") as local_file:
+            local_file.write(file_content)
+            doc = docx.Document(local_file.name)
+            for para in doc.paragraphs:
+                full_text.append(para.text)
+    return '\n'.join(full_text)
+
+def get_text_from_pdf_driveitem(driveitem_object: DriveItem):
+
+    file_content = driveitem_object.get_content().execute_query().value
+    file_text = read_pdf_file(file=io.BytesIO(file_content), file_name=driveitem_object.name)
+    return file_text
+
+def get_text_from_txt_driveitem(driveitem_object: DriveItem):
+    file_content: bytes = driveitem_object.get_content().execute_query().value
+    text_string = file_content.decode('utf-8')
+    return text_string
+
+
+class SharepointConnector(LoadConnector, PollConnector):
+    def __init__(
+        self,
+        batch_size: int = INDEX_BATCH_SIZE,
+        site_list: list[str] = [],
+    ) -> None:
+        self.batch_size = batch_size
+        self.graph_client: GraphClient | None = None
+        self.requested_site_list: list[str] = site_list
+
+    def load_credentials(self, credentials: dict[str, Any]) -> dict[str, Any] | None:
+        aad_client_id = credentials["aad_client_id"]
+        aad_client_secret = credentials["aad_client_secret"]
+        aad_directory_id = credentials["aad_directory_id"]
+
+        def _acquire_token_func():
+            """
+            Acquire token via MSAL
+            """
+            authority_url = f'https://login.microsoftonline.com/{aad_directory_id}'
+            app = msal.ConfidentialClientApplication(
+                authority=authority_url,
+                client_id=aad_client_id,
+                client_credential=aad_client_secret
+            )
+            token = app.acquire_token_for_client(scopes=["https://graph.microsoft.com/.default"])
+            return token
+        
+        self.graph_client = GraphClient(_acquire_token_func)
+        return None
+    
+    def _prune_driveitem_list_by_time(
```

Comment:
Is there a way to add a time range to the query so we don't need to process all drive items through this?